### PR TITLE
Update @resvg/resvg-js to v2.0.0

### PR DESCRIPTION
**Pull request by XhmikosR**
It seems Node.js 12 fails with a  Segmentation fault

@yisibl do you have any ideas?

### CODE REVIEW

**Comment by yisibl** [comment]
I run `npm test` locally and have many test cases failing (node.js 16).
image

**Comment by XhmikosR** [comment]
Weird, it works for me locally with Node.js 16. The only failure I see is the segmentation fault on Node.js 12 on CI.

**Comment by yisibl** [comment]
The main branch (resvg v1.x) doesn't pass even in my local tests. So I can't troubleshoot it now, can you check what recent changes to the test case might have caused it to fail?

**Comment by XhmikosR** [comment]
@yisibl everything works fine here and on CI with resvg 1.x. This branch here only fails with Node.js 12, and the only change is the resvg update to 2.x. :/

**Comment by Kreeg** [comment]
@yisibl mind this https://github.com/puppeteer/puppeteer/issues/8009. Maybe you have the same problem

**Comment by yisibl** [comment]
> mind this https://github.com/puppeteer/puppeteer/issues/8009. Maybe you have the same problem
@Kreeg I've connected the MacBook's power supply and it still doesn't work. Maybe we should look for a more reliable png diff solution.

**Comment by yisibl** [comment]
My guess is that this issue(https://github.com/swc-project/swc/issues/2276#issuecomment-996409481) with Rust may be the cause, @Brooooooklyn is helping to investigate further.

File: lib/svg-sprite/utils/calculate-svg-dimensions.js [hunk 1]
```diff
@@ -17,19 +17,16 @@ const imageSize = require('image-size');
...
     try {
-        const pngData = resvg.render(svg, {
+        const resvg = new Resvg(svg, {
             logLevel: 'error',
             font: {
                 loadSystemFonts: false // It will be faster to disable loading system fonts.
             }
         });
+        const pngData = resvg.render();
+        const { width, height } = imageSize(pngData);
```
[the remark this answers is not in the document]
Comment (XhmikosR) [reply lib/svg-sprite/utils/calculate-svg-dimensions.js:27]: Can you push the removal of `image-size` too? I thought I tried it and it didn't work, but I didn't spend a lot of time :)

[the remark this answers is not in the document]
Comment (XhmikosR) [reply lib/svg-sprite/utils/calculate-svg-dimensions.js:27]: All good, as long as the Node.js 12 issue is fixed and it hits beta or stable, we can update to resvg-js 2.x :)

**Comment by yisibl** [comment]
Please be patient, we are still troubleshooting the cause of the crash, I have 100% reproduced the problem on the Mac, for details of the debugging process please see: https://github.com/yisibl/svg-sprite/pull/3

**Comment by XhmikosR** [comment]
@yisibl I didn't mean to put more pressure on you, sorry about that! I was just stating my plan :)

**Comment by yisibl** [comment]
@XhmikosR It doesn't matter

**Comment by Kreeg** [comment]
@XhmikosR I ran our tests on master on macbook pro late 2013 and they all failed (visually comparing tests). This is critical. I am working on it (might switch to playwright)

**Comment by XhmikosR** [comment]
As long as we can reduce the packages/use system browsers if available, because Playwright is even worse than puppeteer when it comes to disk size :/

**Comment by yisibl** [comment]
@XhmikosR After various debugging, we finally located a possible NAPI bug that was fixed in Node.js 14+, but not cherry-picked to Node.js 12.x.

Please go to support this issue at https://github.com/nodejs/node/issues/42376

**Comment by XhmikosR** [comment]
@yisibl thanks for digging into it! I notice the Node.js developers have already added the needed labels, let's hope they pick the patch before Node.js 12 goes EOL.

**Comment by XhmikosR** [comment]
@yisibl I don't suppose it's possible to add `width` and `height` to the 1.x branch?

**Comment by yisibl** [comment]
@XhmikosR Yes, resvg-js v1.x does not intend to add this API.

**Comment by yisibl** [comment]
Good news, now the last version of Node.js 12 has been released: 12.22.12-rc.0, [PR #42531](https://github.com/nodejs/node/pull/42531). The tests have passed! 🚀🚀🚀

Download: https://nodejs.org/download/rc/v12.22.12-rc.0/

image

https://github.com/yisibl/svg-sprite/runs/5781947632?check_suite_focus=true

**Comment by yisibl** [comment]
In CI you need to change to `setup-node-nvm` to install Node.js 12.22.12-rc.0, please refer to: https://github.com/yisibl/svg-sprite/pull/3/commits/48a402fca42a05608a00be8958430f381ef70721

**Comment by Kreeg** [comment]
@yisibl I just thought the stable release has already been deployed. I think the better way is just to wait for the release

**Comment by yisibl** [comment]
Node.js 12.22.12 already released. https://github.com/nodejs/node/commit/a3d28373d306ed8462a034a8f3019c48354d0cca

**Comment by Kreeg** [comment]
Ok! All tests are passed

**Comment by XhmikosR** [comment]
Let's wait until 12.22.12 has landed in the GitHub images. But this makes me wonder if we should bump the package.json version or not. Probably not, since AFAICT things will work fine unless someone has SVGs without dimensions.

**Comment by XhmikosR** [comment]
@yisibl do you plan to release v2.0.0 final soon?

**Comment by yisibl** [comment]
> @yisibl do you plan to release v2.0.0 final soon?

resvg-js v2.0.0 final is released. https://github.com/yisibl/resvg-js/commit/ecc7a563540e1269cb607102b30e4be9dff76e25

**Review by Kreeg — APPROVED** [review]

**Comment by Kreeg** [comment]
@yisibl that's cool! Thank you!

**Pull request merged**